packages/lesswrong/components/ea-forum/EADigestPage.tsx

```diff
@@ -0,0 +1,296 @@
+      view: "findByNum",
+      num: digestNum
+    },
+    collectionName: "Digests",
+    fragmentName: 'DigestsMinimumInfo',
+    limit: 1,
+    skip: !digestNum
+  })
+  const digest = results?.[0]
+  
+  // get the list of posts in this digest
+  const { data, loading } = useQuery(digestPostsQuery, {
+      ssr: true,
+      skip: !digestNum || !digest,
+      variables: {num: digestNum},
+    }
+  )
+  
+  const onSubscribeClick = async () => {
+    if (currentUser) {
+      try {
+        await updateCurrentUser({
+          subscribedToDigest: true
+        })
+        flash({messageString: (
+          <div className={classes.success}>
+            <ForumIcon icon="CheckCircle" className={classes.successCheckIcon} />
+            <div>
+              Thanks for signing up! You can edit your subscription via
+              your <Link to={'/account?highlightField=subscribedToDigest'} className={classes.successLink}>
+                account settings
+              </Link>.
+            </div>
+          </div>
+        )})
+      } catch(e) {
+        flash('There was a problem subscribing you to the digest. Please try again later.')
+      }
+    }
+  }
+  
+  const {
+    Error404, HeadTags, PostsLoading, EAPostsItem, CloudinaryImage2, ForumIcon, LWTooltip
+  } = Components;
+  
+  // TODO: Probably we'll want to check the publishedDate instead of the endDate, but we haven't been using it.
+  // If we do start using it, we'll need to backfill the publishedDate values and update this condition.
+  const isPublished = digest && digest.endDate
+  
+  // 404 if there is no matching digest, or if the matching one is still in progress and the user is not an admin
+  if ((!digest && !digestLoading) || (!isPublished && !currentUser?.isAdmin)) {
+    return <Error404 />
+  }
+
+  const posts: PostsListWithVotes[] = data?.DigestPosts
+  
+  const postsList = loading ? (
+    <PostsLoading placeholderCount={20} viewType="card" />
+  ) : posts?.map((post) => (
+    <EAPostsItem
+      key={post._id}
+      viewType="card"
+      post={post}
+      showKarma={false}
+    />
+  ))
+  
+  // Define background color with a CSS variable to be accessed in the styles
+  const style = {
+    "--digest-background": digest?.onsitePrimaryColor,
+  } as CSSProperties;
+  
+  // Hide the icon if this is a logged in user who is already subscribed to the digest
+  let subscribeNode = null
+  if (!currentUser || !currentUser.subscribedToDigest) {
+    subscribeNode = <LWTooltip
+      title="Click to subscribe to the weekly email version of this digest"
+      className={classes.subscribeTooltip}
+    >
+      {currentUser ? <ForumIcon
+          icon="Envelope"
+          className={classes.subscribeIcon}
+          onClick={onSubscribeClick}
+        /> : <Link to={digestLink} target="_blank" rel="noopener noreferrer">
+          <ForumIcon icon="Envelope" className={classes.subscribeIcon} />
+        </Link>
+      }
+    </LWTooltip>
+  }
+
+  return (
+    <>
```

Comment:
I know this is the prettier-approved way of doing this but it kills me to waste the indent and extra line on what should obviously be `return <>`

Reply:
Ignore me I am unvirtuous